memory/notes/CombatManager.md

```diff
@@ -0,0 +1,101 @@
+- Spell Locks
+  - for the bomb characters in the demo, breaking a spell lock will kill it - need to find records for that.
+- Loot, did it drop and should we grab it?
+
+Combat Manager
+  - 0xF0 - currentEncounter (Encounter)
+    - 0x110 - encounterDone (bool)
+    - 0x168 - inGameHud (InGameHud)
+      - 0x60 - playerBattleUI (PlayerBattleUI)
+        - 0x50 - comboPointsPanel (ComboPointsPanel)
+          - 0x40 - comboPointerMeter (ComboPointsMeter)
+            - 0x3C - currentComboPoints (float) 0 - 3
+    - 0xE0 - controller
+      - 0x80 - battleUI
+        - 0x40 - playerPanelsList
+          - 0x10 - _items
+            - 0x18 - Count
+            - 0x20 - Item[0] - 0x08 width
+              - 0x28 - hpTextField
+                - 0x50 - currentvalue (the animations current value when losing health)
+                - 0x54 - targetValue (the final value, after animation)
+              - 0x30 - actionPointsTextField (MP)
+                - 0x50 - currentvalue (the animations current value when losing mp)
+                - 0x54 - targetValue (the final value, after animation)
+              - 0x70 - characterDefinitionId (do these vary between reloads?)
+              - 0x78 - selected (is active character for menus)
+
+    - 0x120 - playerActors (Generic.List<PlayerCombatActor>)
+      - 0x10 - _items
+        - 0x18 - count
+        - 0x20 - Item[0] (Player 0)
+          - 0xC8 - dead
+          - 0x150 - fighterDefinition
+            - 0x18 - characterDefinitionId (2851000752) cyclops?
+            - 0x20 - battleCommands
+              - 0x18 - count (5)
+              - 0x20 - Item[0]
+                - 0x18 - descriptionLocId (figure these out)
+        - 0x28 - Item[1] (Player 1)
+          - 0xC8 - dead
+            - 0x150 - fighterDefinition
+              - 0x18 - characterDefinitionId (2851000800) moongirl?
+              - 0x20 - battleCommands
+                - 0x18 - count (5)
+                - 0x20 - Item[0]
+                  - 0x18 - descriptionLocId (figure these out)
+        - 0x30 - Item[2] (Player 2)
+          - 0xC8 - dead
+            - 0x150 - fighterDefinition
+              - 0x18 - characterDefinitionId (2851000848) sunboy?
+              - 0x20 - battleCommands
+                - 0x18 - count (5)
+                - 0x20 - Item[0]
+                  - 0x18 - descriptionLocId (figure these out)
+        - 0x38 - Item[3] (Player 3) ...
+    - 0x130 - enemyTargets (List<EnemyTarget?>)
+      - 0x10 _items
+        - 0x18 count (4?)
+        - 0x20 - item[0]
+          - 0x6C - currentHP
+          - 0x60 - dead (dead enemies seem to disappear from the list)
+        - 0x58 - owner (EnemyController?)
+          - 0xB0 - isPlayer (0)
```

Comment:
This one is unclear. Is the same type structure used for healing abilities too, and this is set to 1 when targetting an ally?

Reply:
isPlayer is simply if they are a player or not - they share a base class with the enemy actors

Reply:
May be a good idea to refactor the code it into `CombatActor` as a base class, with `CombatPlayer` and `CombatEnemy` as inherited. Can wait for later if we find there to be a lot of overlap in the code between the two.

Reply:
I'll look at breaking these up - since technically right now `CombatEnemy` is actually and `EnemyTarget` which is different than the actor class;
I'll rename `CombatEnemy` to `CombatEnemyTarget` and when we use `CombatActor` i'll refactor to a base class